features/events/traps/src/main/java/org/opennms/netmgt/trapd/TrapReceiverSnmp4jImpl.java

```diff
@@ -150,19 +168,93 @@ public void stop() throws InterruptedException {
     /**
      * The execution context.
      */
-    // TODO: HZN-609: Change this to TrapdNotificationListener.receive()
     @Override
     public void run() {
-        SyslogConnection connection = new SyslogConnection(pkt, m_config);
+    }
 
+	@Override
+	public void trapReceived(TrapNotification trapNotification) {
+		try {
+			  for (TrapNotificationHandler handler : m_trapNotificationHandlers) {
+			    handler.handleTrapNotification(trapNotification);
+			  }
+			} catch (Throwable e) {
+			  LOG.error("Handler execution failed in {}", this.getClass().getSimpleName(), e);
+			}
+	}
+
+	@Override
+	public void trapError(int error, String msg) {
+      LOG.warn("Error Processing Received Trap: error = {} {}", error, (msg != null ? ", ref = " + msg : ""));
+	}
+	
+	public void registeredForTraps(){
         try {
```

Comment:
Change this method name to start() and add an @Override annotation to indicate that it overrides an interface method inside TrapReceiver.